crates/pathfinder/src/storage/state.rs

```diff
@@ -1496,6 +1559,47 @@ mod tests {
                 })
             }
         }
+
+        mod interned_version {
+            use super::super::Storage;
+            use super::StarknetBlocksTable;
+
+            #[test]
+            fn duplicate_versions_interned() {
+                let storage = Storage::in_memory().unwrap();
+                let mut connection = storage.connection().unwrap();
+                let tx = connection.transaction().unwrap();
+
+                let blocks = super::create_blocks();
+                let versions = ["0.9.1", "0.9.1"]
+                    .into_iter()
+                    .chain(std::iter::repeat("0.9.2"));
```

Comment:
played it safe, dunno if someone is planning to add more blocks to be returned from `super::create_blocks`.